src/components/Dialog/hooks/useDialog.ts

```diff
@@ -0,0 +1,42 @@
+import { useCallback, useEffect } from 'react';
+import { useDialogManager } from '../../../context';
+import { useStateStore } from '../../../store';
+
+import type { DialogManagerState, GetOrCreateDialogParams } from '../DialogManager';
+
+export const useDialog = ({ id }: GetOrCreateDialogParams) => {
+  const { dialogManager } = useDialogManager();
+
+  useEffect(
+    () => () => {
+      dialogManager.remove(id);
+    },
+    [dialogManager, id],
+  );
+
+  return dialogManager.getOrCreate({ id });
+};
+
+export const useDialogIsOpen = (id: string) => {
+  const { dialogManager } = useDialogManager();
+  const dialogIsOpenSelector = useCallback(
+    ({ dialogsById }: DialogManagerState) => [!!dialogsById[id]?.isOpen],
+    [id],
+  );
+  return useStateStore<DialogManagerState, boolean[]>(dialogManager.state, dialogIsOpenSelector)[0];
```

Comment:
```suggestion
  const dialogIsOpenSelector = useCallback(
    ({ dialogsById }: DialogManagerState) => [!!dialogsById[id]?.isOpen] as const,
    [id],
  );
  return useStateStore(dialogManager.state, dialogIsOpenSelector)[0];
```

Reply:
I wish there was a better way to define tuples in TS but `as const` is as good as it gets. :)